File: lib/models/destiny_milestone_challenge_activity_graph_node_entry.dart

/**  */
class DestinyMilestoneChallengeActivityGraphNodeEntry{
	
	/**  */
	int activityGraphHash;
	
	/**  */
	int activityGraphNodeHash;
	DestinyMilestoneChallengeActivityGraphNodeEntry(
		this.activityGraphHash,
		this.activityGraphNodeHash,
	);

	static DestinyMilestoneChallengeActivityGraphNodeEntry fromMap(Map<String, dynamic> data){
		if(data == null) {
			return null;
		};
		return new DestinyMilestoneChallengeActivityGraphNodeEntry(
				data['activityGraphHash'],
				data['activityGraphNodeHash'],
		);
	}

	static List<DestinyMilestoneChallengeActivityGraphNodeEntry> fromList(List<dynamic> data){
		if(data == null) {
			return null;
		};
		List<DestinyMilestoneChallengeActivityGraphNodeEntry> list = new List();
    data.forEach((item) {
      list.add(DestinyMilestoneChallengeActivityGraphNodeEntry.fromMap(item));
    });
    return list;
	}

	Map<String, dynamic> toMap(){
		Map<String, dynamic> data = new Map();
			data['activityGraphHash'] = this.activityGraphHash;
			data['activityGraphNodeHash'] = this.activityGraphNodeHash;
		return data;
	}
}

File: lib/src/models/DestinyMilestoneChallengeActivityGraphNodeEntry.cs
using System.Runtime.Serialization;

namespace BungieNetApi.Model {
	[DataContract]
	public class DestinyMilestoneChallengeActivityGraphNodeEntry{

		[DataMember(Name="activityGraphHash", EmitDefaultValue=false)]
		public uint ActivityGraphHash { get; set; }

		[DataMember(Name="activityGraphNodeHash", EmitDefaultValue=false)]
		public uint ActivityGraphNodeHash { get; set; }


		public override bool Equals(object input)
        {
            return this.Equals(input as DestinyMilestoneChallengeActivityGraphNodeEntry);
        }

		public bool Equals(DestinyMilestoneChallengeActivityGraphNodeEntry input)
		{
			if (input == null) return false;

			return
				(
                    ActivityGraphHash == input.ActivityGraphHash ||
                    (ActivityGraphHash != null && ActivityGraphHash.Equals(input.ActivityGraphHash))
                ) &&
				(
                    ActivityGraphNodeHash == input.ActivityGraphNodeHash ||
                    (ActivityGraphNodeHash != null && ActivityGraphNodeHash.Equals(input.ActivityGraphNodeHash))
                ) ;
		}
	}
}

